openedx/core/djangoapps/credit/api.py

```diff
@@ -183,14 +183,13 @@ def create_credit_request(course_key, provider_id, username):
 
     """
     try:
-        user_eligibility = CreditEligibility.objects.select_related('course', 'provider').get(
+        user_eligibility = CreditEligibility.objects.select_related('course').get(
             username=username,
-            course__course_key=course_key,
-            provider__provider_id=provider_id
+            course__course_key=course_key
         )
         credit_course = user_eligibility.course
-        credit_provider = user_eligibility.provider
-    except CreditEligibility.DoesNotExist:
+        credit_provider = credit_course.providers.get(provider_id=provider_id)
+    except CreditEligibility.DoesNotExist, CreditProvider.DoesNotExist:
```

Comment:
I don't see the difference between the old and new version except an additional database query.  If the provider doesn't exist, then there won't be any CreditEligibility objects associated with the provider, in which case `UseIsNotEligible` will be raised.  Am I missing something here?